code/_onclick/hud/alerts.dm

```diff
@@ -0,0 +1,204 @@
+			if(thealert.timeout)
+				clear_alert(category)
+				return .()
+			else //no need to update
+				return FALSE
+	else
+		thealert = new type()
+		thealert.override_alerts = override
+		if(override)
+			thealert.timeout = null
+	thealert.mob_viewer = src
+
+	if(new_master)
+		var/old_layer = new_master.layer
+		var/old_plane = new_master.plane
+		new_master.layer = FLOAT_LAYER
+		new_master.plane = FLOAT_PLANE
+		thealert.add_overlay(new_master)
+		new_master.layer = old_layer
+		new_master.plane = old_plane
+		thealert.icon_state = "template" // We'll set the icon to the client's ui pref in reorganize_alerts()
+		thealert.master = new_master
+	else
+		thealert.icon_state = "[initial(thealert.icon_state)][severity]"
+		thealert.severity = severity
+
+	alerts[category] = thealert
+	if(client && hud_used)
+		hud_used.reorganize_alerts()
+	thealert.transform = matrix(32, 6, MATRIX_TRANSLATE)
+	animate(thealert, transform = matrix(), time = 2.5, easing = CUBIC_EASING)
+
+	if(thealert.timeout)
+		addtimer(CALLBACK(src, .proc/alert_timeout, thealert, category), thealert.timeout)
+		thealert.timeout = world.time + thealert.timeout - world.tick_lag
+	return thealert
+
+/mob/proc/alert_timeout(obj/screen/alert/alert, category)
+	if(alert.timeout && alerts[category] == alert && world.time >= alert.timeout)
+		clear_alert(category)
+
+// Proc to clear an existing alert.
+/mob/proc/clear_alert(category, clear_override = FALSE)
+	var/obj/screen/alert/alert = alerts[category]
+	if(!alert)
+		return FALSE
+	if(alert.override_alerts && !clear_override)
+		return FALSE
+
+	alerts -= category
+	if(client && hud_used)
+		hud_used.reorganize_alerts()
+		client.screen -= alert
+	qdel(alert)
+
+/obj/screen/alert
+	icon = 'icons/mob/screen_alert.dmi'
+	icon_state = "default"
+	name = "Alert"
+	desc = "Something seems to have gone wrong with this alert, so report this bug please"
+	mouse_opacity = MOUSE_OPACITY_ICON
+	var/timeout = 0 //If set to a number, this alert will clear itself after that many deciseconds
+	var/severity = 0
+	var/alerttooltipstyle = ""
+	var/override_alerts = FALSE //If it is overriding other alerts of the same type
+	var/mob/mob_viewer //the mob viewing this alert
+
+
+/obj/screen/alert/fire
+	name = "On Fire"
+	desc = "You're on fire. Stop, drop and roll to put the fire out or move to a vacuum area."
+	icon_state = "fire"
+
+/obj/screen/alert/fire/Click()
+	var/mob/living/L = usr
+	if(!istype(L))
+		return
+	L.resist()
+
+//GHOSTS
+//TODO: expand this system to replace the pollCandidates/CheckAntagonist/"choose quickly"/etc Yes/No messages
+/obj/screen/alert/notify_action
+	name = "Notification"
+	desc = "A new notification. You can enter it."
+	icon_state = "template"
+	timeout = 15 SECONDS
+	var/atom/target = null
+	var/action = NOTIFY_JUMP
+
+/obj/screen/alert/notify_action/Click()
+	if(!usr || !usr.client)
+		return
+	if(!target)
+		return
+	var/mob/dead/observer/G = usr
+	if(!istype(G))
```

Comment:
This check could be placed at the start, and if it succeeds there's no need to check for `usr` again.